src/patternfly/components/Drawer/drawer.scss

```diff
@@ -21,55 +33,195 @@
-    --pf-c-drawer__panel--Width: var(--pf-c-drawer__panel--md--Width);
     --pf-c-drawer__panel-body--PaddingTop: var(--pf-c-drawer__panel-body--md--PaddingTop);
     --pf-c-drawer__panel-body--PaddingRight: var(--pf-c-drawer__panel-body--md--PaddingRight);
     --pf-c-drawer__panel-body--PaddingBottom: var(--pf-c-drawer__panel-body--md--PaddingBottom);
     --pf-c-drawer__panel-body--PaddingLeft: var(--pf-c-drawer__panel-body--md--PaddingLeft);
   }
 
+  // Box shadow
+  --pf-c-drawer__panel--BoxShadow: none;
+  --pf-c-drawer--m-expanded__panel--BoxShadow: var(--pf-global--BoxShadow--md-left);
+  --pf-c-drawer--m-expanded--m-panel-left__panel--BoxShadow: var(--pf-global--BoxShadow--md-right);
+
+  // Divider
+  --pf-c-drawer--m-divider__panel--after--Width: var(--pf-global--BorderWidth--sm);
+  --pf-c-drawer--m-divider--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
+
   display: flex;
+  height: 100%;
   overflow-x: hidden;
+}
+
+// Content / panel
+.pf-c-drawer__content,
+.pf-c-drawer__panel {
+  flex-shrink: 0;
+  overflow: auto;
+}
 
-  &.pf-m-expanded {
-    --pf-c-drawer__panel--Transform: var(--pf-c-drawer--m-expanded__panel--Transform);
+// Content
+.pf-c-drawer__content {
+  flex-basis: var(--pf-c-drawer__content--FlexBasis);
+  order: 0;
+}
 
-    @media screen and (min-width: $pf-global--breakpoint--md) {
-      &.pf-m-inline {
-        --pf-c-drawer__content--MarginRight: var(--pf-c-drawer--m-expanded--m-inline__content--MarginRight);
+// Panel
+.pf-c-drawer__panel {
+  position: relative;
+  flex-basis: var(--pf-c-drawer__panel--FlexBasis);
+  order: 1;
+  background-color: var(--pf-c-drawer__panel--BackgroundColor);
+  transition-duration: var(--pf-c-drawer__panel--TransitionDuration);
+  transition-property: var(--pf-c-drawer__panel--TransitionProperty);
+  -webkit-overflow-scrolling: touch;
+}
+
+// Content body
+.pf-c-drawer__content-body.pf-m-padding {
+  padding: var(--pf-c-drawer__content-body--m-padding--PaddingTop) var(--pf-c-drawer__content-body--m-padding--PaddingRight) var(--pf-c-drawer__content-body--m-padding--PaddingBottom) var(--pf-c-drawer__content-body--m-padding--PaddingLeft);
+}
+
+// Panel body
+.pf-c-drawer__panel-body {
+  padding: var(--pf-c-drawer__panel-body--PaddingTop) var(--pf-c-drawer__panel-body--PaddingRight) var(--pf-c-drawer__panel-body--PaddingBottom) var(--pf-c-drawer__panel-body--PaddingLeft);
+}
+
+// Panel body
+.pf-c-drawer__panel-body.pf-m-no-padding {
+  --pf-c-drawer__panel-body--PaddingTop: 0;
+  --pf-c-drawer__panel-body--PaddingRight: 0;
+  --pf-c-drawer__panel-body--PaddingBottom: 0;
+  --pf-c-drawer__panel-body--PaddingLeft: 0;
+}
+
+// Expanded
+.pf-c-drawer.pf-m-expanded .pf-c-drawer__panel {
+  transform: translateX(-100%);
+}
+
+// Layout modifications happen after md breakpoint
+@media screen and (min-width: $pf-global--breakpoint--md) {
+  // Default
+  .pf-c-drawer:not(.pf-m-divider) .pf-c-drawer__panel {
```

Comment:
You could move this down into the `.pf-c-drawer {}` block below with `&:not(.pf-m-divider) .pf-c-drawer__panel`.